src/main/java/io/github/syst3ms/skriptparser/expressions/CondExprIsEmpty.java

```diff
@@ -38,17 +38,17 @@
     @Override
     public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
         expr = expressions[0];
-        isList = matchedPattern == 1;
+        isList = matchedPattern == 0;
         setNegated(parseContext.getParseMark() == 1);
         return true;
     }
 
     @Override
     public boolean check(TriggerContext ctx) {
         var values = expr.getValues(ctx);
-        return isNegated() != (isList
+        return isNegated() != (isList && values.length != 1
                 ? values.length == 0
-                : Arrays.stream(values).allMatch(val -> ((String) val).isBlank()));
+                : Arrays.stream(values).allMatch(val -> val instanceof String && ((String) val).isBlank()));
```

Comment:
If the length of the values is known to be 1, then this can be heavily simplified
```suggestion
                : val instanceof String[] && ((String) val[0]).isBlank());
```